Vale linting & other wording cleanup for KBs (#1209)
returntocorp/semgrep-docs: docs/kb/semgrep-code/finding_all_taints.md

             >> function1(sourceA/sinkB)
```

### Interfile analysis

If both data flows are identified in the same scan, and the scan has interfile analysis enabled (`--pro`, or Pro Engine enabled in the Cloud Platform) only Call stack 1 is reported as a finding. It has a shorter path, and has the same sourceA -> sinkB taint.  

This speeds up triage by ensuring you are only reviewing unique findings. It's especially useful for languages with polymorphic classes that can add noise for a singleton taint.  

### Intrafile analysis

If only intrafile / interprocedural analysis is performed (`--pro-intrafile`), Semgrep only reports a finding for call stack 2. Call stack 1 would not be identified, because it crosses file boundaries.

## Best practices for testing tainted data flows

> If you want to understand in greater detail how Semgrep is detecting tainted data flows, you can use existing or constructed test cases to review the different paths. 

Author: No, this is more of a general statement. Is there a way I could make that clearer/better?

> If you want to understand in greater detail how Semgrep is detecting tainted data flows, you can use existing or constructed test cases to review the different paths. 

Reviewer: ```suggestion
To understand in greater detail how Semgrep detects tainted data flows, you can use your own test cases to review different paths.
```
Removed "existing or constructed" as it adds to the confusion. Existing? Where? From me or Semgrep, Inc?
Also, technically, all test cases are "constructed" so I think we can do without it also. (Just a suggestion).

Author: yeah I agree, I like this revision. It's not letting me apply it because of being outdated, so I'll do it from the CLI and then merge!
